modules/programs/qutebrowser.nix

```
@@ -306,7 +306,17 @@ in {
       };
 
     xdg.configFile."qutebrowser/config.py" =
-      mkIf pkgs.stdenv.hostPlatform.isLinux { text = qutebrowserConfig; };
+      mkIf pkgs.stdenv.hostPlatform.isLinux {
+        text = qutebrowserConfig;
+        onChange = ''
+          hash="$(${pkgs.coreutils}/bin/echo -n $USER | ${pkgs.coreutils}/bin/md5sum | ${pkgs.coreutils}/bin/cut -d' ' -f1)"
+          socket="''${XDG_RUNTIME_DIR:-/run/user/$UID}/qutebrowser/ipc-$hash"
+          if [ -S $socket ]; then
+            ${pkgs.coreutils}/bin/echo "Reloading qutebrowser"
+            $DRY_RUN_CMD ${pkgs.coreutils}/bin/printf '{"args": [":config-source"], "target_arg": null, "protocol_version": 1}\n' | ${pkgs.socat}/bin/socat -lf /dev/null - UNIX-CONNECT:"''${socket}"
```

```suggestion
            $DRY_RUN_CMD echo '{"args": [":config-source"], "target_arg": null, "protocol_version": 1}' | ${pkgs.socat}/bin/socat -lf /dev/null - UNIX-CONNECT:"''${socket}"
```

`printf` is also a shell builtin in many shells.

You can optionally replace the json with `escapeShellArg (builtins.toJSON { args = [ ":config-source" ]; target_arg = null; protocol_version = 1 })`.
Just thought you'd  be interested in this neat trick!